src/main/java/org/icij/datashare/Neo4jUtils.java

```diff
@@ -0,0 +1,722 @@
+                    statement = match.isOptional() ? Cypher.optionalMatch(pattern) :
+                        Cypher.match(pattern);
+                } else {
+                    statement = match.isOptional() ? statement.optionalMatch(pattern) :
+                        statement.match(pattern);
+                }
+            }
+            ExposesReturning returned;
+            if (this.where != null) {
+                returned = statement.where(this.where.into());
+            } else {
+                returned = statement;
+            }
+            Long limit;
+            if (defaultLimit != null) {
+                limit = defaultLimit;
+                if (this.limit != null) {
+                    limit = Math.min(this.limit, defaultLimit);
+                }
+            } else {
+                limit = this.limit;
+            }
+
+            StatementBuilder.OngoingMatchAndReturnWithOrder returnedWithOrder =
+                returned
+                    .returning(Cypher.asterisk())
+                    .orderBy(this.orderBy
+                        .stream()
+                        .map(Neo4jUtils.Into::into)
+                        .collect(Collectors.toList()));
+            if (limit == null) {
+                return returnedWithOrder.build();
+            }
+            return returnedWithOrder
+                .limit(limit)
+                .build();
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) {
+                return true;
+            }
+            if (o == null || getClass() != o.getClass()) {
+                return false;
+            }
+            Query dumpQuery = (Query) o;
+            return java.util.Objects.equals(this.matches, dumpQuery.matches)
+                && java.util.Objects.equals(this.where, dumpQuery.where)
+                && java.util.Objects.equals(this.orderBy, dumpQuery.orderBy)
+                && java.util.Objects.equals(this.limit, dumpQuery.limit);
+        }
+
+        @Override
+        public int hashCode() {
+            return java.util.Objects.hash(matches, where, orderBy, limit);
+        }
+    }
+
+    protected static class SortByProperty implements OrderBy {
+        protected final VariableProperty property;
+        protected final Objects.SortDirection direction;
+
+        @JsonCreator
+        protected SortByProperty(
+            @JsonProperty("property") VariableProperty property,
+            @JsonProperty("direction") Objects.SortDirection direction
+        ) {
+            this.property = java.util.Objects.requireNonNull(property, "missing property");
+            this.direction = java.util.Objects.requireNonNull(direction, "missing sort direction");
+        }
+
+        @Override
+        public SortItem into() {
+            org.neo4j.cypherdsl.core.Property sortByProp = this.property.into();
+            if (this.direction.equals(Objects.SortDirection.ASC)) {
```

Comment:
Since the `Objects.SortDirection` enum has only 2 members and will always have 2, there's no real default, it's one or the other. I'll use a `switch` to make this clearer